src/Components/Slider/Slider.js

```diff
@@ -0,0 +1,90 @@
+import React, { useState, useRef } from 'react';
+import styled from 'styled-components';
+import Product from '../../Pages/Bids/Components/Product';
+
+const Slider = ({ lists, type }) => {
+  const [prevBtnShow, setPrevBtnShow] = useState(false);
+  const [nextBtnShow, setNextBtnShow] = useState(true);
+  const slider = useRef();
+  const prevBtn = useRef();
+
+  const scroll = direction => {
+    let scrollAmount = 0;
+
+    const slide = setInterval(e => {
+      if (direction === 'left') {
+        slider.current.scrollLeft -= 30;
+      } else {
+        slider.current.scrollLeft += 30;
+      }
+      setPrevBtnShow(!!slider.current.scrollLeft);
+      scrollAmount += 5;
+
+      if (scrollAmount >= 100) {
+        setNextBtnShow(!(scrollAmount >= 100 && slider.current.scrollLeft > 0));
+        clearInterval(slide);
+      }
+    }, 20);
+  };
+
+  return (
+    <Wrapper ref={slider}>
+      {prevBtnShow && (
+        <DirectionBtn
+          onClick={() => {
+            scroll('left');
+          }}
+          ref={prevBtn}
+        >
+          <i className="xi-angle-left-min" />
+        </DirectionBtn>
+      )}
+      {lists.map((list, index) => {
+        return type === 'bids' ? <Product key={index} art={list} /> : null;
+      })}
```

Comment:
슬라이더 컴포넌트에서 Product 컴포넌트를 import 하여 사용하는 것은 재사용성의 의미에서 별로 좋지 않은 구조 같습니다. 슬라이더를 사용할 수 있는 경우가 Product 컴포넌트로 이루어진 배열로 제한되기 때문입니다. 이 부분을 children 으로 처리해주면 슬라이더를 더 많은 경우에 사용할 수 있을 것 같습니다.
```suggestion
{children}
```
이렇게 되면, Slider 를 다음과 같이 사용할 것 입니다.
```js
<Slider>
   {lists.map((list, index) => {
      return <Product key={index} art={list} />
   })}
</Slider>
```
이렇게 되면, 어떤 배열에 대해서도 슬라이더를 사용할 수 있게 됩니다.

참고해보세요!

Reply:
넵 이렇게 하는게 더 맞겠네요 감사합니다 😊